### ISSUE TITLE
Slightly confusing error message db_parameter_group syntax fail

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#633]
_This issue was originally opened by @gtmtech as hashicorp/terraform#13014. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Terraform 0.8.7

This message appears:

```
aws_db_parameter_group.db: only lowercase alphanumeric characters and hyphens allowed in "name"
```

When planning an aws_db_parameter_group resource.

However "name" is overloaded and could refer to a parameter name, or resource name. It should be clear, as underscores etc. ARE allowed in parameter names. So the error message refers to the resource "name", not the parameter "name"s

**Comment by bflad** [comment]
This can be easily closed out with adding `parameter group` to:

https://github.com/terraform-providers/terraform-provider-aws/blob/495cf8d0809601a76d6b5369cad05766c3dd90b5/aws/validators.go#L132

**Comment by bflad** [comment]
The error messaging in version 1.27.0 of the AWS provider will be clarified here. 👍

**Comment by bflad** [comment]
This has been released in [version 1.27.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1270-july-11-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
